meteor/client/ui/Settings/ShowStyle/VariantSettings.tsx

```diff
@@ -35,7 +42,101 @@ export const ShowStyleVariantsSettings = withTranslation()(
 
 			this.state = {
 				editedMappings: [],
+				uploadFileKey: Date.now(),
+			}
+		}
+		importVariants(e: React.ChangeEvent<HTMLInputElement>) {
+			const { t } = this.props
+
+			const file = e.target.files ? e.target.files[0] : null
+			if (!file) {
+				return
+			}
+
+			const reader = new FileReader()
+			reader.onload = (e2) => {
+				this.setState({
+					uploadFileKey: Date.now(),
+				})
+
+				const uploadFileContents = (e2.target as any).result
+
+				let newVariants: Array<ShowStyleVariant> = []
+				try {
+					newVariants = JSON.parse(uploadFileContents)
+					if (!_.isArray(newVariants)) {
+						throw new Error('Imported file did not contain an array')
+					}
+				} catch (error) {
+					NotificationCenter.push(
+						new Notification(
+							undefined,
+							NoticeLevel.WARNING,
+							t('Failed to import new showstyle variants: {{errorMessage}}', { errorMessage: error + '' }),
+							'VariantSettings'
+						)
+					)
+					return
+				}
+
+				_.forEach(newVariants, (entry) => {
+					MeteorCall.showstyles
+						.insertShowStyleVariantWithBlueprint(
+							entry.showStyleBaseId,
+							entry.blueprintConfig,
+							entry._id,
+							entry._rundownVersionHash,
+							entry.name
+						)
+						.catch((error) => {
+							NotificationCenter.push(
+								new Notification(
+									undefined,
+									NoticeLevel.WARNING,
+									t('Failed to import Variant ' + entry.name + '. Make sure it is not already imported.', {
+										errorMessage: error + '',
+									}),
+									'VariantSettings'
+								)
+							)
+						})
+				})
 			}
+			reader.readAsText(file)
+		}
+		copyVariant = (showStyleVariant: ShowStyleVariant) => {
+			MeteorCall.showstyles
+				.insertShowStyleVariantWithBlueprint(
+					showStyleVariant.showStyleBaseId,
+					showStyleVariant.blueprintConfig,
+					getRandomId(),
+					'',
+					'Copy of ' + showStyleVariant.name
+				)
+				.catch(console.error)
+		}
+		downloadVariant = (showstyleVariant: ShowStyleVariant) => {
+			const variantArray = [showstyleVariant]
+			const jsonStr = JSON.stringify(variantArray)
+
+			const element = document.createElement('a')
+			element.href = URL.createObjectURL(new Blob([jsonStr], { type: 'application/json' }))
+			element.download = `${showstyleVariant.name}_showstyleVariant_${showstyleVariant._id}.json`
+
+			document.body.appendChild(element) // Required for this to work in FireFox
```

Comment:
Do we want to support Firefox? If not, this could be deleted

Reply:
I added the possibility since everywhere else, where the same functionality is applied, this is taken into consideration.

Reply:
In my opinion we don't have to support Firefox, but it's not my decision :) 
SuperFly might want to support it, but else I would say it's Jacobs decision.